.circleci/config.yml

```diff
@@ -147,21 +181,82 @@ jobs:
+        POSTGRES_DB: postgres
+    working_directory: ~/repo
+    resource_class: large
+
+    environment:
+      JVM_OPTS: -Xmx3200m
+      GRADLE_OPTS: "-Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2"
+      TERM: dumb
+
+    steps:
+    - attach_workspace:
+        at: ~/repo
+
+    - run: apt-get update && apt-get install postgresql-client -y 
 
+    - run: psql -h 127.0.0.1 -p 5432 -U postgres < .docker/initdb.sql
+
+    - run: ./gradlew test --no-daemon
+      
     - run:
-        name: message slack
+        name: Save test results
         command: |
-          export $( cat build/ci.properties | xargs )
-          if [ "${CIRCLE_BRANCH}" = "master" ]
-          then
-            curl -d "token=${SLACK_TOKEN}" \
-            -d "text=EMX2 version: ${TAG_NAME} is released. Check it out: https://emx2.dev.molgenis.net" \
-            -d "channel=C02AZDG6QQ7" \
-            -X POST https://slack.com/api/chat.postMessage          
-          else
-            curl -d "token=${SLACK_TOKEN}" \
-            -d "text=*<${CIRCLE_PULL_REQUEST}|Circle-CI » Molgenis » Molgenis-emx2 » PR-${CIRCLE_PULL_REQUEST##*/} #${CIRCLE_BUILD_NUM}>*
-            PR Preview available on https://preview-emx2-pr-${CIRCLE_PULL_REQUEST##*/}.dev.molgenis.net" \
-            -d "channel=C02AZDG6QQ7" \
-            -X POST https://slack.com/api/chat.postMessage
-          fi
+          mkdir -p ~/test-results/junit/
+          find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} ~/test-results/junit/ \;
+        when: always
+    - store_test_results:
+        path: ~/test-results
+
+
+  notifications:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - run:
+          name: message slack
+          command: |
+            export $( cat build/ci.properties | xargs )
+            if [ "${CIRCLE_BRANCH}" = "master" ]
+            then
+              curl -d "token=${SLACK_TOKEN}" \
+              -d "text=EMX2 version: ${TAG_NAME} is released. Check it out: https://emx2.dev.molgenis.net" \
+              -d "channel=C02AZDG6QQ7" \
+              -X POST https://slack.com/api/chat.postMessage          
+            else
+              curl -d "token=${SLACK_TOKEN}" \
+              -d "text=*<${CIRCLE_PULL_REQUEST}|Circle-CI » Molgenis » Molgenis-emx2 » PR-${CIRCLE_PULL_REQUEST##*/} #${CIRCLE_BUILD_NUM}>*
+              PR Preview available on https://preview-emx2-pr-${CIRCLE_PULL_REQUEST##*/}.dev.molgenis.net" \
+              -d "channel=C02AZDG6QQ7" \
+              -X POST https://slack.com/api/chat.postMessage
+            fi
+
+workflows:
+  version: 2
+  build_and_test:
+    jobs:
+      - prepare
+      - build:
+          requires:
+            - prepare
+      - preview:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore: main
+      - test:
+          requires:
+            - build
+      - notifications:
+          requires:
+            - build
```

Comment:
this is the config part to define a sort of build-test-release graph, see docs for conditions that can be added

Reply:
so issue is that we have 'snapshot' releases and real releases. A snapshot is when a merge into master does NOT include 'feat' or 'fix'. Is there a way to use some other way to have conditions? Because it would be so nice if we can use this workflow mechanism instead of the nasty 'if then else' in shell.